### DISCUSSION TITLE
showing error for using a variable as dimension for a array

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vscode-cpptools#8853] [Q&A]
when i was trying to use a variable in the size of a array .It is indicating a error that is it showing error squiggle . please let me know how to fix it

**Comment by sean-mcmanus** [comment]
We need to know how IntelliSense is being configured. Can you provide the output of running C/C++: Log Diagnostics? Or let us know what OS, compiler, and C or C++ standard is being used? And what version of the C/C++ extension are you using?

This error can occur if the msvc IntelliSense mode is used or the -fms-extensions arg.

**Reply by ghost** [reply]
My os is windows 11 and the compiler is gcc

**Reply by sean-mcmanus** [reply]
We still need more info. What distro and version of gcc? What version of our extension. What language standard? I'm not reproing any issue with `C:\msys64\mingw64\bin\gcc.exe`, gcc 11, c17, with 1.9.0 of our extension. What exactly is the error message? Is "-fms-extensions" being added to the compilerArgs?

**Comment by ghost** [comment]
My gcc version is 9.2.0
The c language standard is c17
The version of your extension is 1.8.4 and the issue is same with the Pre-release Version 1.9.0

**Reply by sean-mcmanus** [reply]
What OS? What is the exact error message? I'm not reproing the issue. The logs from running C/C++: Log Diagnostics (after opening the repro file) may help diagnose what is going wrong.

**Reply by sean-mcmanus** [reply]
Most likely it is using a msvc IntelliSense mode when it shouldn't.

**Reply by ghost** [reply]
Well can you tell me what should i do to fix it and tell me how to run log diagnostics in vs code

**Reply by sean-mcmanus** [reply]
View->Command Palette: C/C++: Log Diagnostics.

**Reply by ghost** [reply]
How to fix this error is there any way by changing settings or by updating ?

**Comment by ghost** [comment]
**This is the output of  log diagnostics**


-------- Diagnostics - 18/2/2022, 12:22:29 pm
Version: 1.8.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.19041.0",
    "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "compilerArgs": [],
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPathIsExplicit": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Users\sai\Documents\c\patterns\starpattern.c ]:
    C:\Users\sai\Documents\c\patterns\starpattern.c
[ C:\Users\sai\Documents\c\patterns\array.c ]:
    C:\Users\sai\Documents\c\patterns\array.c
Translation Unit Configurations:
[ C:\Users\sai\Documents\c\patterns\starpattern.c ]:
    Process ID: 19444
    Memory Usage: 15 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2022\BUILDTOOLS\VC\TOOLS\MSVC\14.30.30705\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c17
    IntelliSense Mode: windows-msvc-x64
[ C:\Users\sai\Documents\c\patterns\array.c ]:
    Process ID: 8624
    Memory Usage: 20 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2022\BUILDTOOLS\VC\TOOLS\MSVC\14.30.30705\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c17
    IntelliSense Mode: windows-msvc-x64
Total Memory Usage: 35 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 4812
Number of files parsed: 13

**Reply by sean-mcmanus** [reply]
Your compilerPath is set to cl.exe. Do you intend to build with that? Can you change your compilerPath to the path to gcc?

**Reply by ghost** [reply]
How could i change that ? **There is no gcc in visual studio build tools**.

Are you saying to change the compiler path to  **c:/Mingw/gcc**. when i changed that to this path it is showing an error in the intellisense mode of c_cpp_properties.json like this **'''IntelliSense mode windows-msvc-x64 is incompatible with compiler path.'''**

**Reply by sean-mcmanus** [reply]
@SaiTeja-Nukala Yeah, also change the IntelliSense mode to windows-gcc-x64 so they match.
